doc/testing.md

```diff
@@ -0,0 +1,9 @@
+# Testing against Flutter
+
+When validating a new lint or experimenting with new language experiments, testing the linter
+against flutter is recommended to help expose false positives and other unexpected behavior.
+
+## Try-Bot testing
```

Comment:
"Try-bot Testing"? Also, I think the steps below ought to be capitalized and punctuated as full sentences.

I think having information like this is great.

I don't know whether this is the easiest/best way to test against flutter, but if it is we probably need to provide a pointer to the docs explaining how to create a CL against the SDK. (They'll have to use Gerrit to choose a try bot.) Also, given that it's often a new rule being tested we probably need to explain how to update the analysis options file used by that bot so that the rule is run.

That said, it might be better to explain how to test rules against other packages and then let users run the rule against the flutter package the way they would against any other package. I think they'd still need to update flutter's analysis options file, which will require a pointer to which file is used, but they might not need to learn Gerrit to do it.

Reply:
I have no idea if this is easiest to be honest...

@a14n : care to add any info on more local testing?

Reply:
To test locally I usually run:

```shell
flutter> bin/cache/dart-sdk/bin/dart $LINTER_HOME/bin/linter.dart --rules my-new-rule dev examples packages | grep '\[lint\]'
```